### DISCUSSION TITLE
renovate.json5: Self-Hosting Renovate vs. GitHub app installation

### DISCUSSION THREAD

**Discussion by hbokh** [renovatebot/renovate#9850] [Request Help] [answered]
Is there a difference for renovate when working with either _Self-Hosting Renovate_ vs. _GitHub app installation_?

I ask because the file at https://github.com/k8s-at-home/template-cluster-k3s/blob/main/.github/renovate.json5 behaves different on a self-hosted setup using `renovate/renovate:slim` version 25.9.0

On self-hosted I get this issue filed by renovate-bot:

---
### Action Required: Fix Renovate Configuration

There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

Location: `renovate.json5`
Error type: The renovate configuration file contains some invalid settings
Message: `packageRules[1]: packageRules cannot combine both matchUpdateTypes and separateMinorPatch. Rule: {"matchDatasources":["docker"],"enabled":true,"commitMessageTopic":"container image {{depName}}","commitMessageExtra":"to {{#if isSingleVersion}}v{{{newVersion}}}{{else}}{{{newValue}}}{{/if}}","matchUpdateTypes":["major","minor","patch"],"separateMinorPatch":true}`

---

Once I comment line 45 https://github.com/k8s-at-home/template-cluster-k3s/blob/main/.github/renovate.json5#L45
in my local repo, things are passing again.

Can anyone explain what's going on here? Thanks in advance!

**Comment by viceice** [comment, marked as answer]
The hosted app is using renovate v24 and you are probably on v25 for the self-hosted slim image.

**Reply by hbokh** [reply]
Indeed. So v25 is stricter?

**Reply by viceice** [reply]
Yes, it errors out for some more wrong config options. The config is also wrong in v24, but renovate hadn't detected it.
